### ISSUE TITLE
Message digest algorithm should be configurable

### ISSUE THREAD

**Issue by pfistererm** [aposin/LicenseScout#84]
The Message digest algorithm used by the LicenseScout should be configurable.

# Feature request

## Expected Behavior - What where you expecting to happen?
Since cryptographic algorithms become compromised over time, it is necessary from time to time to switch to newer algorithms with longer key/hash length or different base algorithms. Therefore it should be possible to select the message digest algorithm used by the LicenseScout as part of the configuration done by the user.

## Current Behavior - What happens?
The MD algorithm is hard-coded to SHA-256. It is not possible to use a newer, more secure algorithm.

## Possible Solution
There should be an additional maven parameter in the configuration section to pass the name of the algorithm. That should look like this:
```
<configuration>
    ...
    <messageDigestAlgorithm>SHA3-512</messageDigestAlgorithm>
    ...
</configuration>
```
The currently used algorithm (SHA-256) should be the default value to keep compatibility.
Therefore, the variable definition in the MOJO should look like this:
```
    /**
     * Algorithm to use for message digests.
     * 
     * <p>The configured name is passed to the Java Cryptography extension (JCE) to obtain an algorithm implementation.
     * Therefore, the name should match an algorithm name supported by one of the configured cryptography providers.</p>
     */
    @Parameter(defaultValue = "SHA-256", property = "messageDigestAlgorithm", required = false)
    private boolean messageDigestAlgorithm;
```
The configured value should be passed to the method org.aposin.licensescout.util.CryptUtil.setMessageDigestAlgorithm(String)

Also, the following code in LicenseCheckedList needs to be adapted:
```
    /**
     * Note: this value is valid for SHA-256, should corresponds to the algorithm returned by {@link CryptUtil#getMessageDigestAlgorithm()}.
     * (Note that the algorithm can be changed by the method {@link CryptUtil#setMessageDigestAlgorithm(String)}, but this length of the hash is always hard-coded.)
     */
    private static final int MESSAGE_DIGEST_NUM_BYTES = 32;
    private static final int MESSAGE_DIGEST_NUM_CODED_CHARACTERS = MESSAGE_DIGEST_NUM_BYTES * 2;
```
## Notes and further information
SHA-256 is still generally considered secure by the cryptographic community. However, a new attack method may change this. In addition, it is a matter of taste/personal view which cryptographic algorithms are preferred.

**Comment by pfistererm** [comment]
The change should be verified with an additional unit test in CryptUtilTest similar to this:
```
    @Test
    public void testCalculateMessageDigest() throws IOException {
CryptUtil.setMessageDigestAlgorithm("SHA3-512");
        final File file = new File("src/test/resources/licensetexts/apache2/LICENSE-2.0-without-newlines.txt");
        final byte[] result = CryptUtil.calculateMessageDigest(file).getBytes();
        final String referenceString = EXPECTED_MESSAGE_DIGEST_STRING_SHA3_512;
        final byte[] reference = getByteArrayFromHexString(referenceString);
        Assert.assertArrayEquals(reference, result);
    }
```
and an additional test dependency (the bouncy castle provider supports SHA3):
```
<dependency>
    <groupId>org.bouncycastle</groupId>
    <artifactId>bcprov-jdk15on</artifactId>
    <version>1.64</version>
    <scope>test</scope>
</dependency>
```

**Comment by pfistererm** [comment]
Closed because implemented in #109 and #112.
